File: lcWeekly42D.py
class Solution(object):
    def replaceWords(self, dict, sentence):
        words = sentence.split(' ')
        maxi=0
        i=0
        while i<len(dict):
            maxi=max(maxi,len(dict[i]))
            i=i+1
        maxi=maxi+1
        i=0
        ansS = ''
        while i<len(words):
            s=words[i]
            j=0
            c=''
            cnt=0
            while j<len(s) and j<maxi:
                c=c+s[j]
                if dict.count(c)>0:
                    ansS=ansS+c
                    cnt=1
                    break
                j=j+1
            if cnt==0:
                ansS=ansS+s
            if i!=len(words)-1:
                ansS=ansS+' '
            i=i+1
        
        return ansS

File: 2017/lcWeekly42D.cpp
#include<stdio.h>
class Solution {
public:
    string replaceWords(vector<string>& dict, string sentence) {
        vector<string> words;
        string token;
        istringstream tokenStream(sentence);
        char delimiter = ' ';
       while (getline(tokenStream, token, delimiter))
       {
          words.push_back(token);
       }
        set<string> mp;
        int cnt;
        for(int i=0;i<dict.size();i++){
            mp.insert(dict[i]);
        }
        string ans="";
        for(int i=0;i<words.size();i++){
            token = words[i];
            string s="";
            cnt=0;
            for(int j=0;j<token.length() && j<101;j++){
                s=s+token[j];
                if(mp.count(s)>0){
                    ans=ans+s;
                    cnt=1;
                    break;
                }
            }
            if(cnt==0){
                ans=ans+token;
            }
            if(i!=words.size()-1){
                ans=ans+" ";
            }
        }
        return ans;
    }
};
